### PR TITLE
Feature/swap jq for glom

### PR DESCRIPTION
**Pull request by matthewpeterkort**
Refs https://github.com/ACED-IDP/aced_etl_pod/issues/17

pyjq does not support python >3.10 

Glom seems to be far better maintained with support for python 3.12: 
https://github.com/mahmoud/glom

This PR swaps out pyjq with glom. pytests pass, graph and simplified schemas match the old schemas that 
were generated with pyjq dependencies. 

Also includes some flake8 fixes to main

### CODE REVIEW

File: iceberg_tools/graph/__init__.py [hunk 1]
```diff
@@ -2,23 +2,38 @@
...
 
-NESTED_OBJECTS = pyjq.compile(
-    '( paths | select(.[-1] == "$ref") ) as $p |   if getpath($p) != "Reference.yaml" then ( $p |  [.[] | tostring]  | join(".") ) else empty end')
-
 NESTED_OBJECTS_IGNORE = ['Identifier', 'Extension']
 
-REFERENCES = pyjq.compile(
-    '( paths | select(.[-1] == "$ref")) as $p |   if getpath($p) == "Reference.yaml" then ( $p | join(".") ) else empty end')
+
+def process_nested_schema_objects(data):
```
Comment (bwalsh) [inline iceberg_tools/graph/__init__.py]: Please add """Comment what this method does."""  docstring.

File: iceberg_tools/graph/__init__.py [hunk 2]
```diff
@@ -2,23 +2,38 @@
...
+    for path in glom(data, "paths", default=[]):
+        if path[-1] == "$ref":
+            path_value = glom.glom(data, path)
+            if path_value != "Reference.yaml":
+                result = ".".join(map(str, path))
+                return result
+    return None
+
+
+def process_nested_schema_references(data):
```
Comment (bwalsh) [inline iceberg_tools/graph/__init__.py]: Please add """Comment what this method does."""  docstring.

File: iceberg_tools/graph/__init__.py [hunk 3]
```diff
@@ -474,9 +489,10 @@ def _extract_values(self, schema_link, vertex) -> list:
         """Extract values from a vertex given a link description object"""
         values = []
         for k, v in schema_link['templatePointers'].items():
-            values_ = self.extract_json_pointer_via_jq(v, vertex)
+            values_ = self.extract_json_pointer_via_glom(v, vertex)
+            print("VALUES: ", values_, type(values_))
```
Comment (bwalsh) [inline iceberg_tools/graph/__init__.py]: remove debugging statement

File: tests/unit/link-description-object/conftest.py [hunk 4]
```diff
@@ -155,7 +154,7 @@ def python_source_directories() -> List[str]:
     - reference: Substance/sub-2
 - id: s-processing-3
   resourceType: Specimen
-  _expected_reference_count: 2
+  _expected_reference_count: 1
```
Comment (bwalsh) [inline tests/unit/link-description-object/conftest.py:157]: Hmm.  Why did reference count change?

File: tests/unit/link-description-object/test_nested_references.py [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
-from iceberg_tools.graph import cast_json_pointer_to_jq
+from iceberg_tools.graph import cast_json_pointer_to_glom
 
 
 def test_nested_references(nested_references: list):
     """Test nested references"""
-    jq = pyjq.compile(cast_json_pointer_to_jq('/processing/-/additive/-/reference'))
+    glom_instance = cast_json_pointer_to_glom('/processing/-/additive/-/reference')
 
+    """
```
Comment (bwalsh) [inline tests/unit/link-description-object/test_nested_references.py:10]: Lets huddle and review

**Review by bwalsh — CHANGES_REQUESTED** [review]
Hey,  nice work finding the glom library.  I was not aware of it.
Couple of items below

File: iceberg_tools/graph/__init__.py (hunk 1 above)
Comment (matthewpeterkort) [reply iceberg_tools/graph/__init__.py]: Those functions probably shouldn't have gotten committed. Was an attempt at a Glom approach to iterate through a nested schema and return all glom paths to $ref fields. 

Didn't end up working and ended up going for a much more conventional approach for traversing a schema.

**Comment by matthewpeterkort** [comment]
Flake8, cleanup Merge ready

**Pull request merged**